src/io_threads.h

```diff
@@ -3,6 +3,10 @@
 
 #include "server.h"
 
+#define TIMESPEC_DIFF_US(start_time, end_time)                                     \
+    (((long long)(end_time).tv_sec - (long long)(start_time).tv_sec) * 1000000LL + \
+     ((long long)(end_time).tv_nsec - (long long)(start_time).tv_nsec) / 1000L)
```

Comment:
`static inline` function is almost always better than macro. We get type checking and better readability. It's still defined in the header file like a macro.

```suggestion
static inline long long timespec_diff_us(struct timespec start_time, struct timespec end_time) {
    return ((long long)end_time.tv_sec - (long long)start_time.tv_sec) * 1000000LL +
           ((long long)end_time.tv_nsec - (long long)start_time.tv_nsec) / 1000LL;
}
```